### PR TITLE
Patron2020 326 system notyfikacji

### PR DESCRIPTION
**Pull request by SzymonFed**

### CODE REVIEW

File: gateway/notifications.h [hunk 1]
```diff
@@ -1,8 +1,13 @@
 #include "dependencies/json.hpp"
+#include "dashboard.h"
 #include <fstream>
+#include <ctime>
+
+#define MAX_ID 10
```
Comment (kuba8686) [inline gateway/notifications.h:6]: not sure how many notifications have to be maintained. Anyway easy to change.

File: gateway/notifications.h [hunk 2]
```diff
@@ -13,31 +18,77 @@ enum type
 };
 
 NLOHMANN_JSON_SERIALIZE_ENUM(type,
+                             {
+                                 {ALERT, "alert"},
+                                 {DOORBELL, "doorbell"},
```
Comment (kuba8686) [inline gateway/notifications.h:23]: we will not have doorbell i think.

File: gateway/notifications.h [hunk 3]
```diff
@@ -51,4 +102,117 @@ class Notifications
...
+                    if (dashboard_previous.smoke_sensors[i].is_smoke_detected != request["isSmokeDetected"].get<bool>())
+                    {
+                        add_notification(create_notification(set_new_id(), dashboard.smoke_sensors[i].id));
+                        dashboard_previous.smoke_sensors[i].is_smoke_detected = request["isSmokeDetected"].get<bool>();
+                        return 0;
+                    }
+                }
+            }
+        }
+
```
Comment (kuba8686) [inline gateway/notifications.h:178]: What about HVAC? Finally i dont know if HVAC status change have to be included at notificatiions or not.

Comment (SzymonFed) [reply gateway/notifications.h:178]: Had to cut it unintentionally, added it in next commit.

File: gateway/notifications.h (hunk 2 above)
Comment (SzymonFed) [reply gateway/notifications.h:23]: Hmm, it's probably a leftover from earlier versions. Will remove it later.

**Pull request closed without merging**